network-config-analyzer/PeerContainer.py

```diff
@@ -257,7 +257,19 @@ def _add_pod_from_yaml(self, pod_object):
             owner_name = owner_references[0].get('name', '')  # take the first owner
             owner_kind = owner_references[0].get('kind', '')
 
-        pod = Pod(pod_name, pod_namespace, owner_name, owner_kind)
+        service_account_name = ''
+        if kind == 'Pod':
+            # serviceAccountName for a Pod:
+            # TODO: should we use 'default' here?
+            service_account_name = pod_object['spec'].get('serviceAccountName', 'default')
+        elif kind in {'Deployment', 'ReplicaSet', 'StatefulSet', 'DaemonSet', 'Job', 'CronJob',
+                      'ReplicationController'}:
```

Comment:
We should define somewhere in the class the set of all pod-creating resources. Currently it is replicated 4 times.

Reply:
done